core/metacling/src/TClingCallbacks.h

```diff
@@ -83,6 +86,8 @@ class TClingCallbacks : public cling::InterpreterCallbacks {
    //
    virtual void TransactionCommitted(const cling::Transaction &T);
 
+   virtual void beforeEmittingModuleForTransaction(const cling::Transaction &T);
```

Comment:
ModuleEmission is easily confused with CodeGen emitting an `llvm::Module`. Please clarify this in the function name.